sw/airborne/firmwares/fixedwing/stabilization/stabilization_adaptive.c

```diff
@@ -520,3 +596,107 @@ inline static void h_ctl_pitch_loop(void)
   h_ctl_elevator_setpoint = TRIM_PPRZ(cmd);
 }
 
+/* yaw damper */
+#if H_CTL_YAW_LOOP
+inline static void h_ctl_yaw_loop(void)
+{
+ 
+#if H_CTL_YAW_TRIM_BY
+  // Actual Acceleration from IMU:
+ #ifndef SITL
+  struct Int32Vect3 accel_meas_body;
+  struct Int32RMat *body_to_imu_rmat = orientationGetRMat_i(&imu.body_to_imu);
+  int32_rmat_transp_vmult(&accel_meas_body, body_to_imu_rmat, &imu.accel);
+  float by = ACCEL_FLOAT_OF_BFP(accel_meas_body.y);
+ #else
+  float by = 0;
+ #endif
+
+  if (pprz_mode == PPRZ_MODE_MANUAL || launch == 0) {
+    h_ctl_yaw_by_sum_err = 0.;
+  } else {
+     if (h_ctl_yaw_by_igain > 0.){
+//     only update wehn: phi<60degrees and by<2g
+       if(fabsf(stateGetNedToBodyEulers_f()->phi) < 1.05 &&
+          fabsf(by) < 20.) {
+       h_ctl_yaw_by_sum_err += by * H_CTL_REF_DT;
+//     max half rudder deflection for trim
+       BoundAbs(h_ctl_yaw_by_sum_err, MAX_PPRZ/(2* h_ctl_yaw_by_igain));
+       }
+   } else {
+      h_ctl_yaw_by_sum_err = 0.;
+    }
+  }
+#endif
+
+#ifdef USE_AIRSPEED
+float Vo = *stateGetAirspeed_f();
+Bound(Vo, STALL_AIRSPEED, RACE_AIRSPEED);
+#else
+float Vo = NOMINAL_AIRSPEED;
+#endif
+
+  h_ctl_ref.yaw_rate = h_ctl_yaw_rate_setpoint // set by RC
+                       + 9.81f/ Vo * sinf(h_ctl_roll_setpoint) // for turns
+;
+  float d_err = h_ctl_ref.yaw_rate - stateGetBodyRates_f()->r;
+
+  float cmd = + h_ctl_yaw_dgain * d_err
+#if H_CTL_YAW_TRIM_BY
+              - h_ctl_yaw_by_igain * h_ctl_yaw_by_sum_err
+#endif
+;
+  cmd /= airspeed_ratio2;
+  h_ctl_rudder_setpoint = TRIM_PPRZ(cmd);
+}
+#endif
+
+/* CL with Flaps - direct lift control */
+#if H_CTL_CL_LOOP
+inline static void h_ctl_cl_loop(void)
+{
+
+#ifndef SITL
+  struct Int32Vect3 accel_meas_body;
+  struct Int32RMat *body_to_imu_rmat = orientationGetRMat_i(&imu.body_to_imu);
+  int32_rmat_transp_vmult(&accel_meas_body, body_to_imu_rmat, &imu.accel);
+  float bx = ACCEL_FLOAT_OF_BFP(accel_meas_body.z);
```

Comment:
Not sure I get what you want to do. Is it x or z ?
Can you explain a little bit the CL control ?